src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,16 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, that's the terminology
+            //we always used with users, although we prefer inline internally
+            if (this == INLINE) {
```

Comment:
I see the confusion... have a look at this discussion: https://github.com/elastic/elasticsearch/pull/10116#discussion_r26744097 . @clintongormley @s1monw can you guys come to an agreement? :) 
Using inline would simplify things but I don't like the fact that we change terminology for our users to just simplify our code. I am more for renaming the ScriptType.INLINE to DYNAMIC later on on master and leave this as-is for now.